staramr/blast/JobHandler.py

```diff
@@ -93,8 +94,38 @@ def run_blasts(self, files) -> None:
         db_files = self._make_db_from_input_files(self._input_genomes_tmp_dir, files)
         logger.debug("Done making blast databases for input files")
 
+        future_mlst_db = []
         logger.info("Scheduling MLST for input files")
-        self._schedule_mlst(db_files)
+        max_files_length = len(db_files)
+
+        partition = math.floor(max_files_length/self._threads)
+
+        currStart = 0
+        currEnd = 0
+
+        # Partition the files only if it's greater than the number of threads available
```

Comment:
Your right, after testing it again and implementing your suggestions, turns out I didn't need it at all. I decided to call the thread pool during the `db_files` and added a check to make sure `self._thread_pool_executor` is not `None` to avoid linting problems

Reply:
Awesome. Thanks.